cmd/erasure-zones.go

```diff
@@ -179,7 +205,13 @@ func (z *erasureZones) getZoneIdx(ctx context.Context, bucket, object string, op
 		// Success case and when DeleteMarker is true return.
 		return i, nil
 	}
-	return z.getAvailableZoneIdx(ctx), nil
+
+	// We multiply the size by 2 to account for erasure coding.
+	idx = z.getAvailableZoneIdx(ctx, size*2)
+	if idx < 0 {
+		return -1, errDiskFull
```

Comment:
We should return a higher-level error here `toObjectErr(errDiskFull)`  would be needed